packages/vhd-lib/src/merge.js

```diff
@@ -23,109 +24,91 @@ export default limitConcurrency(2)(async function merge(
 ) {
   const mergeStatePath = dirname(parentPath) + '/' + '.' + basename(parentPath) + '.merge.json'
 
-  const parentFd = await parentHandler.openFile(parentPath, 'r+')
-  try {
-    const parentVhd = new VhdFile(parentHandler, parentFd)
-    const childFd = await childHandler.openFile(childPath, 'r')
-    try {
-      const childVhd = new VhdFile(childHandler, childFd)
-
-      let mergeState = await parentHandler.readFile(mergeStatePath).catch(error => {
-        if (error.code !== 'ENOENT') {
-          throw error
-        }
-        // no merge state in case of missing file
-      })
-
-      // Reading footer and header.
-      await Promise.all([
-        parentVhd.readHeaderAndFooter(
-          // dont check VHD is complete if recovering a merge
-          mergeState === undefined
-        ),
+  return await Disposable.use(async function* () {
```

Comment:
FYI, latest `promise-toolbox` version has a `Disposable.wrap` to allow a whole function to use disposables.